### DISCUSSION TITLE
RWMol not filling in hydrogens at stereocenter

### DISCUSSION THREAD

**Discussion by diethylmalonate** [rdkit/rdkit#4873] [Q&A]
When bonds are broken at a stereocenter using RemoveBond, implicit hydrogens aren't added even when the stereocenter is no longer asymmetrical. Code example:

`from rdkit import Chem`
`m = Chem.MolFromSmiles("O[C@](Cl)(F)C")`
`molW = Chem.RWMol(m)`
`molW.RemoveBond(0,1)`
`molW.RemoveBond(1,2)`
`Chem.SanitizeMol(molW)`
`print(Chem.MolToSmiles(molW))`
`>>> C[C]F.Cl.O`

is there a way to force-fill the missing hydrogens to obtain `CCF.Cl.O`?

Thanks

**Comment by ptosco** [comment]
For each bond you break you need to re-fill the emptied valences with Hs, one at each end of the bond:

```
from rdkit import Chem
from rdkit.Chem.Draw import IPythonConsole
IPythonConsole.drawOptions.addAtomIndices = True

m = Chem.MolFromSmiles("O[C@](Cl)(F)C")
m
```
![image](https://user-images.githubusercontent.com/5244385/149346351-e5f0914f-e558-47e1-a3ee-5512a353009f.png)

```
molW = Chem.RWMol(m)
molW.RemoveBond(0, 1)
molW.RemoveBond(1, 2)
molW.GetAtomWithIdx(0).SetNumExplicitHs(molW.GetAtomWithIdx(0).GetNumExplicitHs() + 1)
molW.GetAtomWithIdx(1).SetNumExplicitHs(molW.GetAtomWithIdx(1).GetNumExplicitHs() + 2)
molW.GetAtomWithIdx(2).SetNumExplicitHs(molW.GetAtomWithIdx(2).GetNumExplicitHs() + 1)
Chem.SanitizeMol(molW)
rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_NONE
molW
```
![image](https://user-images.githubusercontent.com/5244385/149344623-3535f555-cc99-4dac-ba96-d42f9e9da4d8.png)

```
print(Chem.MolToSmiles(molW))
CCF.Cl.O
```

You will also probably want to reassign stereochemistry to get rid of the chiral tag which should not be there anymore since you removed the asymmetry by breaking those two bonds:
```
for a in molW.GetAtoms():
    print(a.GetIdx(), a.GetSymbol(), a.GetChiralTag())
0 O CHI_UNSPECIFIED 
1 C CHI_TETRAHEDRAL_CCW
2 Cl CHI_UNSPECIFIED 
3 F CHI_UNSPECIFIED 
4 C CHI_UNSPECIFIED 
Chem.AssignStereochemistry(molW, True, True)
molW
```
![image](https://user-images.githubusercontent.com/5244385/149345876-0e6516b5-5cf0-46e2-9d8c-a85ec22797f0.png)

```
print(Chem.MolToSmiles(molW))
CCF.Cl.O
for a in molW.GetAtoms():
    print(a.GetIdx(), a.GetSymbol(), a.GetChiralTag())
0 O CHI_UNSPECIFIED 
1 C CHI_UNSPECIFIED 
2 Cl CHI_UNSPECIFIED 
3 F CHI_UNSPECIFIED 
4 C CHI_UNSPECIFIED

**Comment by bp-kelley** [comment]
The function: Chem.FragmentOnBonds also does this if you don't add dummy
atoms:

>>> Chem.MolToSmiles(Chem.FragmentOnBonds(Chem.MolFromSmiles("O[C@](Cl)(F)C"),
[0,1], False))
'CCF.Cl.O'
